omod-2.0/src/main/java/org/openmrs/module/attachments/rest/AttachmentResource2_0.java

```diff
@@ -26,4 +49,109 @@ public Attachment save(Attachment delegate) {
+		
+		if (!obs.isComplex()) {
+			throw new IllegalRequestException("The following obs is not a complex obs, no complex data can be retrieved. "
+			        + "Obs UUID: " + obs.getUuid());
+		}
+		
+		ComplexViewHelper viewHelper = context.getComplexViewHelper();
+		String view = new String();
+		Obs complexObs = Context.getObsService().getComplexObs(obs.getObsId(), viewHelper.getView(obs, view));
+		ComplexData complexData = complexObs.getComplexData();
+		
+		// setup attachment instance
+		attachment = new Attachment();
+		attachment.setComplexData(complexData);
+		
+		// Switching to our complex data object
+		ValueComplex valueComplex = new ValueComplex(obs.getValueComplex());
+		AttachmentComplexData attComplexData = context.getComplexDataHelper().build(valueComplex.getInstructions(),
+		    complexData);
+		
+		// generate the bytes array from the AttachmentComplexData
+		byte[] complexByte = attComplexData.asByteArray();
+		
+		// fetch the mime type of the bytes array
+		InputStream is = new BufferedInputStream(new ByteArrayInputStream(complexByte));
+		try {
+			URLConnection.guessContentTypeFromStream(is);
+		}
+		catch (IOException e) {
+			e.printStackTrace();
+		}
+		
+	}
+	
+	@PropertyGetter("bytesMimeType")
+	public static String getBytesMimeType(Attachment attachment) {
+		return attachment.getBytesMimeType();
+	}
+	
+	/**
+	 * Sets the bytes contentFamily of an attachment
+	 * 
+	 * @param the attachment instance
+	 * @param contentFamily of the bytes to set
+	 */
+	@PropertySetter("bytesContentFamily")
+	public void setBytesContentFamily(Attachment attachment, ContentFamily bytesContentFamily) {
+		Obs obs = new Obs();
+		obs = context.getObsService().getObsByUuid(obs.getUuid());
+		
+		if (!obs.isComplex()) {
+			throw new IllegalRequestException("The following obs is not a complex obs, no complex data can be retrieved. "
+			        + "Obs UUID: " + obs.getUuid());
+		}
+		
+		ComplexViewHelper viewHelper = context.getComplexViewHelper();
+		String view = new String();
+		Obs complexObs = Context.getObsService().getComplexObs(obs.getObsId(), viewHelper.getView(obs, view));
+		ComplexData complexData = complexObs.getComplexData();
+		
+		// setup attachment instance
+		attachment = new Attachment();
+		attachment.setComplexData(complexData);
+		
+		// Switching to our complex data object
+		ValueComplex valueComplex = new ValueComplex(obs.getValueComplex());
+		AttachmentComplexData attComplexData = context.getComplexDataHelper().build(valueComplex.getInstructions(),
+		    complexData);
+		byte[] complexByte = attComplexData.asByteArray();
+		
+		// convert the byte array into a string
+		String str = new String(complexByte, StandardCharsets.UTF_8);
+		
+		// fetch the content family
+		context.getContentFamily(str);
+	}
+	
+	@PropertyGetter("bytesContentFamily")
+	public static ContentFamily getBytesContentFamily(Attachment attachment) {
+		return attachment.getBytesContentFamily();
+	}
```

Comment:
I don't think we need all this, you can just set the two new members of `Attachment` after [this line](https://github.com/gitcliff/openmrs-module-attachments/blob/b012ca9cbb1a95c999e45767a72b5b120f7d0bed/api/src/main/java/org/openmrs/module/attachments/obs/Attachment.java#L53).

Also, since you will need [`ComplexDataHelper`](https://github.com/openmrs/openmrs-module-attachments/blob/9c61cf896be64bb09544287a567140df387b97dd/api/src/main/java/org/openmrs/module/attachments/obs/ComplexDataHelper.java#L16) anyway, you may as well do all this in `AttachmentResource1_10`.
That will probably mean that the [constructor `Attachment(Obs)`](https://github.com/gitcliff/openmrs-module-attachments/blob/b012ca9cbb1a95c999e45767a72b5b120f7d0bed/api/src/main/java/org/openmrs/module/attachments/obs/Attachment.java#L38) will have to become `Attachment(Obs, ComplexDataHelper)`, or something along these lines.

Reply:
@mks-d how about fetching `ComplexDataHelper ` using context rather than to alter the `constructor Attachment(Obs) ` ?,,,,,,

Reply:
@gitcliff no let's avoid those decoupled uses of `Context`, just make it a constructor parameter for now.